src/Percentiler.h

```diff
@@ -0,0 +1,97 @@
+#ifndef GCP_PERCENTILER_H
+#define GCP_PERCENTILER_H
+
+#include <iostream>
+#include <map>
+#include <vector>
+#include <algorithm>
+#include <stdexcept>
+
+#include "custom_types.h"
+
+
+class NotEnoughData : public std::runtime_error {
+  using std::runtime_error::runtime_error;
+};
+
+/**
+ * Class responsible for assigning confidence percentiles to raw genotype confidences.
+ * This class is decoupled from the others (to satisfy https://github.com/leoisl/GCP/issues/6)
+ */
+class Percentiler {
+public:
+  /**
+   * Builds a GenotypeConfidencePercentiler from a vector of simulated genotype confidences
+   */
+  explicit Percentiler(const std::vector<GenotypeConfidence> &input_entries) :
+          entries(input_entries),
+          min_entry(*std::min_element(input_entries.begin(), input_entries.end())),
+          max_entry(*std::max_element(input_entries.begin(), input_entries.end())) {
+
+      bool enough_data = input_entries.size() >= 2;
+      if (not enough_data) {
+          throw NotEnoughData("Please provide at least two simulated genotype confidences.");
+      }
+  }
+
+
+  /**
+   * Get the confidence percentile given a genotype confidence.
+   */
+  GenotypePercentile get_confidence_percentile(GenotypeConfidence query) {
+      auto lo = std::lower_bound(entries.begin(), entries.end(), query);
```

Comment:
but since we're calling std::lower_bound, isnt that self documenting? to be honest, i just picked lo and hi from python code for the `bisect` function.

Reply:
i changed to `lower_bound`